src/base/bittorrent/session.cpp

```diff
@@ -4513,6 +4514,43 @@ void Session::startUpTorrents()
         if (needStore)
              m_resumeDataStorage->store(torrentID, resumeData);
 
+        const QString category = resumeData.category;
+        bool isCategoryRecovered = recoveredCategories.contains(category);
+        if (!category.isEmpty() && (isCategoryRecovered || !m_categories.contains(category)))
+        {
+            if (!isCategoryRecovered)
+            {
+                if (addCategory(category))
+                {
+                    recoveredCategories.insert(category);
+                    isCategoryRecovered = true;
+                    LogMsg(tr("Inconsistent data is detected. Category '%2' is assigned to some torrent(s) but doesn't exist. It will be recovered with default settings.")
+                           .arg(torrentID.toString(), category), Log::WARNING);
+                }
+                else
+                {
+                    resumeData.category.clear();
+                    LogMsg(tr("Inconsistent data is detected. Torrent '%1' has assigned category '%2' that is invalid.")
```

Comment:
>Torrent '%1' has assigned category '%2' that is invalid

Maybe `Torrent '%1' has assigned to category '%2' which is invalid`
Other contributors may have better suggestions..

Reply:
All done.
>Other contributors may have better suggestions..

Fine-tuning of log messages isn't so important for this PR but fixing crash is. If someone does not have time to express their opinion about current messages, they can be changed later.

Reply:
`Torrent '%1' was assigned to category '%2' which is now invalid`

`Torrent '%1' previously assigned to category '%2' is invalid`

Reply:
We are talking here about a situation where there is an incorrect category name in torrent "resume data", which we cannot create because of this.

Reply:
What about `Category '%2' assigned to torrent '%1' is invalid`?